pyteal/compiler/compiler_test.py

```diff
@@ -2195,2122 +2199,264 @@ def access_b4_store(magic_num: pt.abi.Uint64, *, output: pt.abi.Uint64):
+    @pt.ABIReturnSubroutine
+    def log_1(*, output: pt.abi.Uint64) -> pt.Expr:
+        return output.set(1)
+
+    meth = router.add_method_handler(
+        log_1,
+        method_config=pt.MethodConfig(
+            no_op=pt.CallConfig.CALL,
+            opt_in=pt.CallConfig.CALL,
+        ),
+    )
 
-// sub
-sub_1:
-store 10
-store 9
-load 9
-load 10
--
-store 11
-load 11
-retsub
+    assert meth.method_signature() == "log_1()uint64"
 
-// mul
-mul_2:
-store 16
-store 15
-load 15
-load 16
-*
-store 17
-load 17
-retsub
+    @pt.ABIReturnSubroutine
+    def log_creation(*, output: pt.abi.String) -> pt.Expr:
+        return output.set("logging creation")
 
-// div
-div_3:
-store 22
-store 21
-load 21
-load 22
-/
-store 23
-load 23
-retsub
+    meth = router.add_method_handler(
+        log_creation, method_config=pt.MethodConfig(no_op=pt.CallConfig.CREATE)
+    )
+    assert meth.method_signature() == "log_creation()string"
 
-// mod
-mod_4:
-store 28
-store 27
-load 27
-load 28
-%
-store 29
-load 29
-retsub
 
-// all_laid_to_args
-alllaidtoargs_5:
-store 63
-store 62
-store 61
-store 60
-store 59
-store 58
-store 57
-store 56
-store 55
-store 54
-store 53
-store 52
-store 51
-store 50
-store 49
-store 48
-load 48
-load 49
-+
-load 50
-+
-load 51
-+
-load 52
-+
-load 53
-+
-load 54
-+
-load 55
-+
-load 56
-+
-load 57
-+
-load 58
-+
-load 59
-+
-load 60
-+
-load 61
-+
-load 62
-+
-load 63
-+
-store 64
-load 64
-retsub
+ON_COMPLETION_ACTIONS = pt.BareCallActions(
+    opt_in=pt.OnCompleteAction.call_only(pt.Log(pt.Bytes("optin call"))),
+)
 
-// empty_return_subroutine
-emptyreturnsubroutine_6:
-byte "appear in both approval and clear state"
-log
-retsub
+FIRST_ROUTER = pt.Router(
+    "ASimpleQuestionablyRobustContract",
+    ON_COMPLETION_ACTIONS,
+    clear_state=pt.Approve(),
+)
+add_methods_to_router(FIRST_ROUTER)
 
-// log_1
-log1_7:
-int 1
-store 66
-load 66
-retsub
 
-// log_creation
-logcreation_8:
-byte 0x00106c6f6767696e67206372656174696f6e
-store 68
-load 68
-retsub""".strip()
+def router_app_tester() -> tuple[list[pt.Router], dict[str, str]]:
+    routers = []
+    sources = {}
 
-    assert expected_ap_with_oc == actual_ap_with_oc_compiled
+    def append_router_info(rinfo, programs):
+        assert len(rinfo) == 3
+        assert len(programs) == 2
+        routers.append(rinfo)
+        sources[rinfo[:2]] = programs
 
-    expected_csp_with_oc = """#pragma version 6
-int 1
-return""".strip()
-    assert expected_csp_with_oc == actual_csp_with_oc_compiled
+    # V6 not ready for Frame Pointers:
+    with pytest.raises(pt.TealInputError) as e:
+        pt.Router("will-error", ON_COMPLETION_ACTIONS).compile_program(
+            version=6, optimize=pt.OptimizeOptions(frame_pointers=True)
+        )
+    assert "Frame pointers aren't available" in str(e.value)
 
+    def router_compilarison(router, version, fixture_ap, fixture_csp, key, write=False):
+        (actual_ap, actual_csp, _) = router.compile_program(version=version)
```

Comment:
nit: can omit a tuple
```suggestion
        actual_ap, actual_csp, _ = router.compile_program(version=version)
```